src/stdlib_linalg_schur.fypp

```diff
@@ -0,0 +1,358 @@
+        
+        contains
+        
+            ! Interface to dummy select routine
+            pure logical(lk) function do_not_select(alpha#{if rt.startswith('r')}#r,alphai#{endif}#) 
+                ${rt}$, intent(in) :: alpha#{if rt.startswith('r')}#r,alphai#{endif}# 
+                do_not_select = .false.
+            end function do_not_select 
+        
+    end subroutine get_schur_${ri}$_workspace   
+
+    ! Schur decomposition subroutine
+    module subroutine stdlib_linalg_${ri}$_schur(a,t,z,eigvals,overwrite_a,storage,err)
+        !> Input matrix a[m,m]
+        ${rt}$, intent(inout), target :: a(:,:)
+        !> Schur form of A: upper-triangular or quasi-upper-triangular matrix T
+        ${rt}$, intent(out), contiguous, target :: t(:,:)
+        !> Unitary/orthonormal transformation matrix Z
+        ${rt}$, optional, intent(out), contiguous, target :: z(:,:)
+        !> [optional] Output eigenvalues that appear on the diagonal of T
+        complex(${rk}$), optional, intent(out), contiguous, target :: eigvals(:)
+        !> [optional] Provide pre-allocated workspace, size to be checked with schur_space
+        ${rt}$, optional, intent(out), target :: storage(:)
+        !> [optional] Can A data be overwritten and destroyed?
+        logical(lk), optional, intent(in) :: overwrite_a        
+        !> [optional] State return flag. On error if not requested, the code will stop
+        type(linalg_state_type), optional, intent(out) :: err
+
+        ! Local variables
+        integer(ilp) :: m,n,mt,nt,ldvs,nvs,lde,lwork,sdim,info
+        logical(lk) :: overwrite_a_
+        logical(lk), target :: bwork_dummy(1),local_eigs
+        logical(lk), pointer :: bwork(:)
+        real(${rk}$), allocatable :: rwork(:)
+        ${rt}$, target :: vs_dummy(1,1)
+        ${rt}$, pointer :: vs(:,:),work(:),eigs(:)#{if rt.startswith('r')}#,eigi(:)#{endif}#
+        character :: jobvs,sort
+        type(linalg_state_type) :: err0
+        
+        ! Problem size
+        m  = size(a, 1, kind=ilp)
+        n  = size(a, 2, kind=ilp)        
+        mt = size(t, 1, kind=ilp)
+        nt = size(t, 2, kind=ilp)
+        
+        ! Validate dimensions
+        if (m/=n .or. m<=0 .or. n<=0) then
+            err0 = linalg_state_type(this, LINALG_VALUE_ERROR, 'Matrix A must be square: size(a)=',[m,n])
+            call linalg_error_handling(err0, err)
+            return
+        end if    
+        if (mt/=nt .or. mt/=n .or. nt/=n) then
+            err0 = linalg_state_type(this, LINALG_VALUE_ERROR, 'Matrix T must be square: size(T)=',[mt,nt], &
+                                                          'should be',[m,n])
+            call linalg_error_handling(err0, err)
+            return
+        end if                
+        
+        !> Copy data into the output array
+        t = a 
+        
+        ! Can A be overwritten? By default, do not overwrite
+        if (present(overwrite_a)) then 
```

Comment:
```Fortran
overwrite_a_ = .false._lk
if (present(overwrite_a)) overwrite_a_ = overwrite_a .and. n>=2
```
Suggestion just to keep the style already in use elsewhere

Reply:
[d5e4019](https://github.com/fortran-lang/stdlib/pull/892/commits/d5e4019d407834767e8359ace69bac2c3a553f88)